frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,16 +1,91 @@
-import { Button } from '~/libs/components/components.js';
+import hidePasswordIcon from '~/assets/img/hidePassword.svg';
+import showPasswordIcon from '~/assets/img/showPassword.svg';
+import { Button, Input, Link } from '~/libs/components/components.js';
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useAppForm, useCallback, useState } from '~/libs/hooks/hooks.js';
+import {
+  type UserSignInRequestDto,
+  userSignInValidationSchema,
+} from '~/packages/users/users.js';
+import { DEFAULT_LOGIN_PAYLOAD } from '~/pages/auth/components/sign-in-form/libs/common/constants.js';
+
+import styles from './styles.module.scss';
 
 type Properties = {
-  onSubmit: () => void;
+  onSubmit: (payload: UserSignInRequestDto) => void;
 };
 
-const SignInForm: React.FC<Properties> = () => (
-  <>
-    <h1>Sign In</h1>
-    <form>
-      <Button label="Sign in" />
-    </form>
-  </>
-);
+const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
+  const { control, errors, handleSubmit } = useAppForm({
+    defaultValues: DEFAULT_LOGIN_PAYLOAD,
+    validationSchema: userSignInValidationSchema,
+  });
+  const [isPasswordVisible, setPasswordVisibility] = useState(false);
+
+  const handleTogglePasswordVisibility = useCallback((): void => {
+    setPasswordVisibility((previousValue) => !previousValue);
+  }, [setPasswordVisibility]);
+
+  const handleFormSubmit = useCallback(
+    (event_: React.BaseSyntheticEvent): void => {
+      void handleSubmit(onSubmit)(event_);
+    },
+    [handleSubmit, onSubmit],
+  );
+
+  return (
+    <div className={styles.wrapper}>
+      <h2 className={styles.title}>Hello!</h2>
+      <form
+        className={styles.form}
+        name="loginForm"
+        onSubmit={handleFormSubmit}
+      >
+        <fieldset className={styles.fieldset}>
+          <Input
+            name="email"
+            type="email"
+            placeholder="Enter your email"
+            label="Email"
+            control={control}
+            errors={errors}
+          />
+          <div className={styles.passwordInputWrapper}>
+            <button
+              type="button"
+              className={styles.passwordToggle}
+              onClick={handleTogglePasswordVisibility}
+            >
+              <img
```

Comment:
Please add @todo comment here
```suggestion
              /**
               * @todo
               * replace with icon component after corresponding PR will be merged
             */
              <img
```

Reply:
**UPDATE:** replace it with Icon component